.github/workflows/cd.yaml

```diff
@@ -38,14 +37,98 @@ jobs:
-          ZQD_DATA_URI: s3://zqd-demo-1/mark/zqd-meta
-          ZQD_K8S_USER: ci-master
-          ZQD_TEST_CLUSTER: zq-test.us-east-2.eksctl.io
+          ZQD_DATA_URI: s3://zq-ci-master/${{ env.TODAY }}.${{ github.run_number }}
         run: |
-          aws eks --region us-east-2 update-kubeconfig --name zq-test
-          kubectl config set-context --current --namespace=ci-master
+          set -x
+          aws eks --region us-east-2 update-kubeconfig --name $CLUSTER
+          kubectl create ns $NAMESPACE || :
+          kubectl config set-context --current --namespace=$NAMESPACE
+          make helm-uninstall
+          make helm-install-recruiter
+          make helm-install-root
+          make helm-install-worker
+  eks-test-zapi-1:
+    # Run smoke tests for zqd image prior to deploying to ci-master.
+    needs: eks-test-deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-2
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Set current date as env variable
+        run: echo "TODAY=$(date +'%y-%m-%d')" >> $GITHUB_ENV
+      - env:
+          CLUSTER: zq-test
+          NAMESPACE: ci-${{ github.run_number }}
+          SOURCE_LOG_1: s3://brim-sampledata/wrccdc/zeek-logs/files.log.gz
+          SPACE: files
+          ZQD_DATA_URI: s3://zq-ci-master/${{ env.TODAY }}.${{ github.run_number }}
+        run: |
+          set -x
+          make install
+          aws eks --region us-east-2 update-kubeconfig --name $CLUSTER
+          kubectl config set-context --current --namespace=$NAMESPACE
+          kubectl scale --replicas=4 deployment/worker-zqd
+          kubectl port-forward svc/recruiter-zqd 8020:9867 &
+          kubectl port-forward svc/root-zqd 9867:9867 &
+          sleep 2 # wait for port-forwards to complete
+          zapi ls
+          zapi new -k archivestore -d $ZQD_DATA_URI $SPACE
+          # Post an archive of about 1.6 GB to run test queries.
+          zapi -s $SPACE post $SOURCE_LOG_1
+          # There should be 4 workers running now.
+          [[ 4 == $(curl http://localhost:8020/recruiter/listfree | jq -r ".workers" | jq length) ]]     
+          time zapi -s $SPACE get -workers 1 -t '"FgzbkQ15TOoOENn1lb" | sort tx_hosts'
+          time zapi -s $SPACE get -workers 2 -t '"FgzbkQ15TOoOENn1lb" | sort tx_hosts'
+          time zapi -s $SPACE get -workers 4 -t '"FgzbkQ15TOoOENn1lb" | sort tx_hosts'
+          time zapi -s $SPACE get -workers 4 -t 'count()' > count.tzng
+          time zapi -s $SPACE get -workers 4 -t \
+            '* | count() by tx_hosts | sort -r count | head 5' > countbytxhost.tzng
```

Comment:
The last two zapi queries are slow at the moment. We should fix that. Since they use 4 workers to do a full scan of the archive, they should be able to run at the same speed as the 4 worker search, about 10 seconds. But right now they take over 90 seconds. When we fix that these tests will finish sooner.